pkg/clients/argocd.go

```diff
@@ -80,11 +80,18 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 	if err != nil {
 		return nil, err
 	}
+	grpcWeb := false
+	if pc.Spec.GRPCWeb != nil {
+		grpcWeb = *pc.Spec.GRPCWeb
+	}
```

Comment:
Can be simplied with https://pkg.go.dev/k8s.io/utils@v0.0.0-20230726121419-3b25d923346b/ptr#Deref

```suggestion
	grpcWeb := ptr.Deref(pc.Spec.GRPCWeb, false)
```